packages/medusa/src/services/shipping-option.ts

```diff
@@ -556,19 +683,25 @@ class ShippingOptionService extends BaseService {
         }
       }
 
-      if ("amount" in update && option.price_type !== "calculated") {
+      if (
+        "amount" in update &&
+        update.amount &&
+        option.price_type !== "calculated"
+      ) {
         option.amount = update.amount
       }
 
-      if ("name" in update) {
+      if ("name" in update && update.name) {
         option.name = update.name
       }
 
-      if ("admin_only" in update) {
+      if ("admin_only" in update && update.admin_only) {
```

Comment:
This is a boolean so add an undefined check - maybe we should make this a clear convention so we never get tripped up by similar situations?